### PR TITLE
Refresh metastore on Lambda searcher

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Fix outdated metastore issues on the search side
- add configurable polling interval on searcher
- change polling ticker to `DELAY` to avoid bursts after longer Lambda inactivity

Also:
- adding some docs summarizing all configurable env variables
- made indexer timeout configurable in CDK script to fix the HDFS indexer example work out of the box
- some refactoring/comments on the LazyFileBackedIndex to help readability

### How was this PR tested?

Intensive testing on our cloud account.

### CODE REVIEW

File: quickwit/quickwit-metastore/src/metastore/file_backed/lazy_file_backed_index.rs [hunk 1]
```diff
@@ -29,9 +29,8 @@ use tracing::error;
 use super::file_backed_index::FileBackedIndex;
 use super::store_operations::{load_index, METASTORE_FILE_NAME};
 
-/// Lazy [`FileBackedIndex`]. It loads a `FileBackedIndex`
-/// on demand and optionally spawns a task to poll
-/// regularly the storage and update the index.
+/// Lazy [`FileBackedIndex`]. It loads a `FileBackedIndex` on demand. When the index is first
+/// loaded, it optionally spawns a task to regularly poll the storage and update the index.
```
Comment (guilload) [inline quickwit/quickwit-metastore/src/metastore/file_backed/lazy_file_backed_index.rs:33]:
```suggestion
/// loaded, it optionally spawns a task to periodically poll the storage and update the index.
```

File: quickwit/quickwit-metastore/src/metastore/file_backed/lazy_file_backed_index.rs [hunk 2]
```diff
@@ -68,15 +67,24 @@ impl LazyFileBackedIndex {
         }
     }
 
-    /// Get `FileBackedIndex`.
+    /// Get a syncronized `FileBackedIndex`. If the index wasn't provided on creation, we load it
```
Comment (guilload) [inline quickwit/quickwit-metastore/src/metastore/file_backed/lazy_file_backed_index.rs:70]:
```suggestion
    /// Gets a synchronized `FileBackedIndex`. If the index wasn't provided on creation, we load it
```

File: quickwit/quickwit-metastore/src/metastore/file_backed/lazy_file_backed_index.rs [hunk 3]
```diff
@@ -122,6 +130,7 @@ fn spawn_index_metadata_polling_task(
 ) {
     tokio::task::spawn(async move {
         let mut interval = tokio::time::interval(polling_interval);
+        interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
```
Comment (guilload) [inline quickwit/quickwit-metastore/src/metastore/file_backed/lazy_file_backed_index.rs:133]: TIL

**Comment by rdettai** [comment]
This PR is still failing when creating a new index with:
```
internal error: could not find checkpoint for index `hdfs-logs:XXXXXXXX` and source `_ingest-lambda-source`
```

Tried to create the `_ingest-lambda-source` with `CreateIndexRequest`, but that is not possible because `identifiers must match the following regular expression: ^[a-zA-Z][a-zA-Z0-9-_\.]{2,254}$`.

**Pull request merged**